pep-0722.rst

```diff
@@ -74,42 +73,41 @@ commonly-used approach.
 Specification
 =============
 
-Any Python script may contain one or more *metadata blocks*. Metadata blocks are
-identified by reading the script *as a text file* (i.e., the file is not parsed
-as Python source code), looking for contiguous blocks of lines that start with
-the identifying characters ``##``. Whitespace is not allowed before the
-identifying ``##``. More than one metadata block may exist in a Python file.
+The content of this section will be published in the Python Packaging user
+guide, PyPA Specifications section, as a document with the title "Embedding
+Metadata in Script Files".
 
-Tools reading metadata blocks MAY respect the standard Python encoding
-declaration. If they choose not to do so, they MUST process the file as UTF-8.
+Any Python script may contain a *dependency block*. The dependency block is
+identified by reading the script *as a text file* (i.e., the file is not parsed
+as Python source code), looking for the first line of the form::
 
-Within a metadata block, whitespace after the ``##`` and at the end of the line
-is ignored, and blank lines are ignored. The first line of a metadata block is
-special, and identifies the type of block. This line MUST contain a colon
-character, ``:``. If the colon is not present, the block is not considered to be
-a metadata block, and tools MUST ignore it. The block type is all of the text on
-the initial line, up to the colon. There must be no whitespace before the colon.
-The initial line MAY contain text after the colon. How this is interpreted
-depends on the block type. Block types MUST be treated as case sensitive.
+   # Script Dependencies:
 
-The interpretation of any lines in a metadata block after the initial
-identifying line, is defined by the type of block.
+The hash character must be at the start of the line with no preceding whitespace.
+The text "Script Dependencies" is recognised regardless of case, and the spaces
+represent arbitrary whitespace (although at least one space must be present). The
+following regular expression recognises the dependency block header line::
 
-Tools MUST ignore any blocks with types they do not handle.
+    (?i)^#\s+script\s+dependencies:\s*$
 
-Block types starting with the characters ``X-`` are reserved for the user, and
-MUST NOT be given a meaning in any future standard.
+Tools reading the dependency block MAY respect the standard Python encoding
+declaration. If they choose not to do so, they MUST process the file as UTF-8.
 
-Otherwise, the only defined block type is ``Script Dependencies``. For this
-block type,
+After the header line, all lines in the file up to the first line that doesn't
+start with a ``#`` sign are considered *dependency lines* and are treated as
+follows:
 
-1. Text after the colon on the initial line is NOT allowed.
-2. All subsequent lines MUST contain :pep:`508` requirement
-   specifiers, one per line.
+1. The initial ``#`` sign is stripped.
```

Comment:
According to the spec, yes it is. I don't care enough to prohibit it, but black will reformat it to add a space (and flake8 will complain about it) so I doubt it will be common.

It's not particularly important to me to *enforce* readability, just to *allow* it. People can do stupid things, but that's fine. For example, I'm never going to be able to prevent

```
# Script Dependencies:
#
#    numpy # This ensures pandas is installed, because we use click to parse the command line
```

So why worry about a missing space, that's still perfectly readable?